vespalib/src/tests/net/tls/openssl_impl/openssl_impl_test.cpp

```diff
@@ -632,7 +632,8 @@ TEST_F("Failure statistics are incremented on authorization failures", CertFixtu
     EXPECT_EQUAL(0u, client_stats.invalid_peer_credentials);
     EXPECT_EQUAL(1u, server_stats.failed_tls_handshakes);
     EXPECT_EQUAL(0u, server_stats.tls_connections);
-    EXPECT_EQUAL(0u, client_stats.tls_connections);
+    // Client TLS connection count may be 0 (<= v1.2) or 1 (v1.3), since v1.3
+    // completes its handshake earlier.
```

Comment:
will this double-count connections as both successful and unsuccessful in metrics?

Reply:
It shouldn't. Based on the underlying compiled-in OpenSSL version the handshake shall either complete or fail.

Reply:
will this instead mean that we will be unable to detect handshake issues using the handshake-related metrics since connections prematurely succeeds to later fail... ?

Reply:
Ah sorry, I understand what you mean now. Yes, for a client it's likely that we'll end up upping both the handshake OK metric and the broken connections metric right afterwards. Though we should explicitly test this to make sure (for instance, maybe we break the connection from the server before we can send an alert frame at all) and to see if we can find a workaround.

The current set of authorization-related unit tests do not bother waiting for alert frames to be sent/received at all, just for the handshakes themselves to complete or fail.